cpu/cc2538/periph/uart.c

```diff
@@ -251,7 +252,11 @@ void uart_write(uart_t uart, const uint8_t *data, size_t len)
                bootstrapped */
             /* If Tx FIFO is full then add to ring buffer */
             if (uart_config[uart].dev->cc2538_uart_fr.FRbits.TXFF) {
-                while (tsrb_add_one(&uart_tx_rb[uart], data[i]) < 0) {}
+                while (tsrb_add_one(&uart_tx_rb[uart], data[i]) < 0) {
+                    /* Wait until the buffer is empty */
+                    mutex_trylock(&uart_ctx[uart].tx_empty);
+                    mutex_lock(&uart_ctx[uart].tx_empty);
```

Comment:
Why are you locking the mutex twice?

Reply:
The first call is to lock the mutex if it isn't already. The second is to wait for the unlock in the TX interrupt.